test/test_packet_id_allocator.js

```diff
@@ -0,0 +1,181 @@
+import test from 'ava'
+import { PacketIdAllocator } from '../dist/utils/packetIdAllocator.js'
+
+const maxPacketId = 65535
+
+test.todo('all insertion cases')
+
+test('should allocate packet IDs in increasing order', async (t) => {
+  t.plan(maxPacketId);
+  const allocator = new PacketIdAllocator();
+
+  for (let i = 1; i <= maxPacketId; ++i) {
+    const id = await allocator.allocate();
+    t.assert(i === id);
+  }
+})
+
+test('should end up with one interval after releasing all IDs in random order', async (t) => {
+  t.plan(1)
+  const allocator = new PacketIdAllocator();
+
+  /* Allocate all IDs */
+  while (await allocator.allocate() < maxPacketId);
+
+  /* [1, 2, 3, ..., 65535] */
+  const idsToRelease = Array.from({ length: maxPacketId }, (_, i) => i + 1);
+  /* Shuffle array using Fisher-Yates */
```

Comment:
why shuffle the array?

Reply:
Testing that regardless of the insertion order the final result is correct